lib/resources/iptables.rb

```diff
@@ -43,29 +43,26 @@ def initialize(params = {})
   end
 
   def has_rule?(rule = nil, _table = nil, _chain = nil)
-    found = false
-    retrieve_rules.each { |line|
-      # checks if the rule is part of the ruleset
-      # for now, we expect an excact match
-      found = true if line.casecmp(rule) == 0
-    }
-    found
+    # checks if the rule is part of the ruleset
+    # for now, we expect an exact match
+    retrieve_rules.any? { |line| line.casecmp(rule) == 0 }
   end
 
   def retrieve_rules
-    return @iptables_cache if defined?(@iptables_cache)
+    return @iptables_cache if @iptables_cache
```

Comment:
although ruby implicitly does the right thing, using `defined?` on a variable that has not been defined is better at expressing why this line is used